workers/unity/Packages/io.improbable.gdk.core/Subscriptions/CallbackManagers/GuardedCallbackManagerSet.cs

```diff
@@ -24,12 +24,15 @@ public void AddCallbackManager(TIndex index, TManager manager)
 
         public List<TManager> GetManagers()
         {
-            foreach (var indexToAdd in indexesToAdd)
+            if (indexesToAdd.Count > 0)
```

Comment:
is this needed? does not checking it actually increase the overhead that much?

Reply:
Clearing already empty lists does have an overhead. We have other places were we have this exact pattern (if not empty, iterate, clear). So I've made sure this follows the same pattern.